ariadne/cardano/src/Ariadne/Wallet/Backend/KeyStorage.hs

```diff
@@ -387,89 +278,62 @@ generateMnemonic entropySize = do
 -- | Generate a mnemonic and a wallet from this mnemonic and add the
 -- wallet to the storage.
 newWallet ::
-       AcidState DB
+       PassiveWalletLayer IO
     -> WalletConfig
     -> WalletFace
-    -> (CardanoMode ~> IO)
     -> PassPhrase
     -> Maybe WalletName
     -> Maybe Byte
     -> IO [Text]
-newWallet acidDb walletConfig face runCardanoMode pp mbWalletName mbEntropySize = do
+newWallet pwl walletConfig face pp mbWalletName mbEntropySize = do
   let entropySize = fromMaybe (wcEntropySize walletConfig) mbEntropySize
   mnemonic <- generateMnemonic entropySize
-  -- The empty string below is called a passphrase in BIP-39, but
-  -- it's essentially an extra mnemonic word. We consider it an
-  -- advanced feature and do not provide it for now.
-  -- generateMnemonic function guarantees that mnemonic is a non-empty list
-  let seed = mnemonicToSeed (unwords $ Unsafe.init mnemonic) ""
-  let (_, esk) = safeDeterministicKeyGen seed pp
-  hasPass <- mkHasPass runCardanoMode (pp == emptyPassphrase)
-  mnemonic <$
-    addWallet acidDb face runCardanoMode esk mbWalletName mempty hasPass assurance
+  let seed = mnemonicToSeedNoPassword $ unwords $ Unsafe.init mnemonic
+      (_, esk) = safeDeterministicKeyGen seed pp
+  mnemonic ++ ["ariadne-v0"] <$
```

Comment:
In `generateMnemonic` I already append `"ariadne-v0"`. Please sync this code with master.